sdk/tables/azure-data-tables/azure/data/tables/aio/_table_client_async.py

```diff
@@ -671,17 +669,17 @@ async def submit_transaction(
             **kwargs,
         )
         try:
-            for operation in operations:  # type: ignore
-                batched_requests.add_operation(operation)
-        except TypeError:
-            try:
-                async for operation in operations:  # type: ignore
+            if hasattr(operations, "__iter__"):
```

Comment:
I think you can actually use `isinstance(Iterable)`/`isinstance(AsyncIterable)` as well.

Reply:
Ah, just found `isinstance(Iterable)`/`isinstance(AsyncIterable)` works in current mypy 1.4.1. Will update.